include/private/entry.h

```diff
@@ -199,22 +395,80 @@ strbuilder_append_procid( struct strbuilder *builder );
  *
  * **Async Cancel Safety: AC-Unsafe heap**
  * This function is not safe to call from threads that may be asynchronously
- * cancelled, as the memory allocation function may not be AC-Safe.
+ * cancelled, as the memory allocation function may not be AC-Safe./**
+ * Locks the mutex within the entry.
  *
- * @param builder The strbuilder to append the characters to.
+ * **Thread Safety: MT-Safe**
+ * This function is thread safe.
  *
- * @param entry The entry to extract the structured data from.
+ * **Async Signal Safety: AS-Unsafe **
+ * This function is not safe to call from signal handlers due to the use of
+ * mutexes that are not guarenteed to be async signal safe.
+ *
+ * **Async Cancel Safety: AC-Unsafe**
+ * This function is not safe to call from threads that may be asynchronously
+ * cancelled, due to the use of lock that could be left in undefined state.
  *
- * @return The modified builder if no error is encountered. If an error is
+ * @param entry The entry to be locked.
+ *
+ */
+void
+lock_entry( const struct stumpless_entry *entry );
```

Comment:
looks like there are multiple declarations of `lock_entry` now? This should be resolved.